kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/model/networking/OpenShiftRoutesClusterIngressNetworkingModel.java

```diff
@@ -0,0 +1,135 @@
+                                                           java.util.List<NodeIdRanges> nodeIdRanges,
+                                                           Tls tls,
+                                                           int sharedSniPort)
+        implements ClusterIngressNetworkingModel, SharedLoadBalancerServiceRequirements {
+
+    public OpenShiftRoutesClusterIngressNetworkingModel {
+        Objects.requireNonNull(cluster);
+        Objects.requireNonNull(ingress);
+        Objects.requireNonNull(proxy);
+        Objects.requireNonNull(nodeIdRanges);
+        Objects.requireNonNull(openShiftRoutes);
+        Objects.requireNonNull(tls);
+    }
+
+    private static final int ROUTE_TLS_PORT = 443;
+
+    public static final int DEFAULT_CLIENT_FACING_LOADBALANCER_PORT = 9083;
+
+    @Override
+    public Stream<ServiceBuilder> services() {
+        return Stream.empty();
+    }
+
+    @Override
+    public Stream<ContainerPort> identifyingProxyContainerPorts() {
+        return Stream.empty();
+    }
+
+    @Override
+    public NodeIdentificationStrategy nodeIdentificationStrategy() {
+        String bootstrap = bootstrapServersHost();
+        String advertisedBrokerAddressPattern = prefixedClusterIngressHost("broker-$(nodeId)");
+        return new SniHostIdentifiesNodeIdentificationStrategy(HostPort.asString(bootstrap, sharedSniPort),
+                HostPort.asString(advertisedBrokerAddressPattern, ROUTE_TLS_PORT));
+    }
+
+    public String prefixedClusterIngressHost(String prefix) {
+        String baseDomain = name(cluster) + "-" + name(ingress) + "." + openShiftRoutes.getIngressDomain();
+        return prefix + "." + baseDomain;
+    }
+
+    @Override
+    public Stream<RouteBuilder> routes() {
+        RouteBuilder bootstrapRoute = routeBuilder(name(cluster) + "-" + name(ingress) + "-bootstrap", bootstrapServersHost());
+        IntStream upstreamNodeIds = nodeIdRanges.stream()
+                .flatMapToInt(nodeIdRanges -> IntStream.rangeClosed(Math.toIntExact(nodeIdRanges.getStart()),
+                        Math.toIntExact(nodeIdRanges.getEnd())));
+        var upstreamNodeRoutes = upstreamNodeIds
+                .mapToObj(nodeId -> routeBuilder(name(cluster) + "-" + name(ingress) + "-" + nodeId, prefixedClusterIngressHost("broker-" + nodeId)));
+        return Stream.concat(Stream.of(bootstrapRoute), upstreamNodeRoutes);
+    }
+
+    private RouteBuilder routeBuilder(String routeName, String host) {
+        return new RouteBuilder().withNewMetadata().withName(routeName).withNamespace(namespace(proxy))
+                .withOwnerReferences(ResourcesUtil.newOwnerReferenceTo(proxy), ResourcesUtil.newOwnerReferenceTo(ingress), ResourcesUtil.newOwnerReferenceTo(cluster))
+                .withLabels(Labels.standardLabels(proxy))
+                .endMetadata()
+                .withNewSpec().withHost(host)
+                .withNewPort().withTargetPort(new IntOrString(sharedSniPort)).endPort()
```

Comment:
this tripped me, the Route takes the port of the pod, not the port exposed by the Service. Our Service maps `9081 -> 9291` but the Route needs to be configured with 9291.